test/e2e_node/dynamic_kubelet_config_test.go

```diff
@@ -908,6 +924,11 @@ func updateConfigMapFunc(f *framework.Framework, tc *nodeConfigTestCase) error {
 	if err != nil {
 		return err
 	}
+	// update the ResourceVersion of the expected config with the version from the newly set one if
+	// we expect it to succeed and to be set
+	if tc.expectedConfigUpdateSuccess && tc.expectConfig != nil && tc.expectConfigStatus.lastKnownGood != nil {
```

Comment:
just question: I guess `expectedConfigUpdateSuccess: false` means `beforeKC(or lkgKC)` is the same as `expectConfig`, right?

Reply:
afaik. `beforeKC` is just the config of the kubelet in the beginning of the tests.

`expectedConfigUpdateSuccess: false` means that we don't expect the applied configmap to be correctly set as the kubeletconfig, and therefore end up as the lastKnownGood-configmap.

That can be a result of:
- The config isn't valid/doesn't work, and when applied kubelet fails, and the config is rolled back. Then the `lastKnownGood`-configmap is then applied.
- The config is similar as the previous one. Then the ResourceVersion of `lastKnownGood` will not update, it will keep the previous one. https://github.com/kubernetes/kubernetes/blob/1089cb629f6a9ff89ceb2af4caff42fde466047e/pkg/kubelet/kubeletconfig/controller.go#L310-L314


If that make sense?